MemoryMeter.c

```diff
@@ -27,6 +27,7 @@ static void MemoryMeter_updateValues(Meter* this) {
    char* buffer = this->txtBuffer;
    size_t size = sizeof(this->txtBuffer);
    int written;
+   int simple_used;
```

Comment:
1. Use the `double` type. Don't use `int` as it could overflow (> 4 GiB).
2. I suggest the name `unavailable` here. It's difficult to recognize what `simple_used` means.
```suggestion
   double unavailable;
```

Reply:
Good catch on the type. Not sure what I was thinking there.

`used` is what that number's called in the help text, so I think it should match that. But that is itself an issue, because then wouldn't be the same as the other `used` displays. But then that's also been the case in the past - it did used to be effectively "unavailable", based on the `available` value, but that was changed in favour of using a value based on the bars.

I think `unavailable` would also be a bit off, suggesting an association with `available` that isn't directly there.

How about `unreclaimable`? Or, simpler, just `claimed`? Get rid of the negative.

That's effectively I'm trying to distinguish here. The claimed bars we can't easily displace (except to swap) - used and shared - versus the scavenging bars that will yield - cache and buffers.

This would be a UI question first - would we be okay `claimed/total]` in the help?

Reply:
Arguably, the problem starts with using the word `used` for only a fraction of the memory. Special areas have been carved out making `used` ever more specific.  But really `used` is the opposite of `free` - all 4 bars are `used`.

It's probably too late to do a full rework now, but more consistent naming would be

    |unshared|shared|buffers|cache|free|
    |-----claimed---|-reclaimable-|free|
    |-------------used------------|free|
    |---------------total--------------|

At least we can use `claimed` here.

Reply:
The order is unshared, buffers, shared, cache, and free at the moment.

I'm impartial about the actual wording, but claimed seems fine with me, although this might cause yet another confusion because of yet another terminology introduced that's possibly in use elsewhere already.

Reply:
> The order is unshared, buffers, shared, cache, and free at the moment.

Indeed, but that diagram shows the conceptual problem with that.  

Arguably there might be other ways of grouping the memory, but I can't actually think of one that justifies the current order. `shared` is associated with `used` by being unreclaimable - just a shared/unshared distinction.  `buffers` is associated with `cache` by being disk cache - a block/file distinction - and being reclaimable generally.  (Actually Linux's reclaimable slab is hiding inside `cache` at the minute - not disk, but reclaimable).

So separating `shared` from `used` and `buffers` from `cache` loses two associations, and I can't see anything gained.

It's just an error arising from the `shared` split not being initially fully considered. Original state was:

    |"used"-|buffers|cache|free|
    |claimed|-reclaimable-|free|
    |---------used--------|free|
    |-----------total----------|

`shared` was new, so just defaulted to the end, meaning it was wrongly separated from `used`.

    |unshared|buffers|cache|shared-|free|
    |claimed-|-reclaimable-|claimed|free|   !!! claimed discontiguous  !!! reclaimable not last
    |-------------used-------------|free|
    |---------------total---------------|

#566 noted that it wasn't reclaimable, and reclaimable should be last, so moved it left one. It got closer to `used` (good, but not perfect), and separated `buffers` from `cache` (bad).

    |unshared|buffers|shared-|-cache-|free|
    |claimed-|reclaim|claimed|reclaim|free|   !!! totally discontiguous
    |-------------used---------------|free|
    |---------------total-----------------|

Reordering shared to second just gets us back to where we started.

Reply:
> I'm impartial about the actual wording, but claimed seems fine with me, although this might cause yet another confusion because of yet another terminology introduced that's possibly in use elsewhere already.

Just as you wrote this I pushed an update with `claimed`. It feels good to me. I can see Linux using "reclaimable" and "unreclaimable", so it uses "unreclaimable" for the same concept. But that feels a bit too heavy if used in isolation - they have uses of "reclaimable" so it's the natural counterpart. We don't have any uses of "reclaimable".

Reply:
Updated visualisations now that "compressed" is in there too.

Before this patch:

```
|unshared|buffers|shared|compressed|-cache-|free|
|--used--|----unused----|---used---|---unused---|  <- discontiguous regions count as "used" in the label
|---------------------total---------------------|
```

After this patch
```
|unshared|shared|compressed|buffers|cache|free|
|--working set--|-inactive-|----unclaimed-----|
|---------claimed----------|----unclaimed-----|  <- contigious regions count as "claimed" in the label
|---------------------total-------------------|
```

Last chart shows justification for having shared to the left of compressed - it's grouping together the "live" working set.